src/hooks/useLiveChartData.ts

```diff
@@ -1,216 +1,216 @@
-    fetchCoingeckoDataSWR,
-    {
-      shouldRetryOnError: false,
-      revalidateOnFocus: false,
-      revalidateIfStale: false,
-    },
-  )
-
-  const chartData = useMemo(() => {
-    if (!isKyberDataNotValid && kyberData && kyberData.length > 0) {
-      return kyberData
-        .sort((a: any, b: any) => parseInt(a.timestamp) - parseInt(b.timestamp))
-        .map((item: any) => {
-          return {
-            time: parseInt(item.timestamp) * 1000,
-            value: !isReverse ? item.token0Price : item.token1Price || 0,
-          }
-        })
-    } else if (coingeckoData && coingeckoData[0]?.prices?.length > 0 && coingeckoData[1]?.prices?.length > 0) {
-      const [data1, data2] = coingeckoData
-      return data1.prices.map((item: number[]) => {
-        const closestPrice = getClosestPrice(data2.prices, item[0])
-        return { time: item[0], value: closestPrice > 0 ? parseFloat((item[1] / closestPrice).toPrecision(6)) : 0 }
-      })
-    } else return []
-  }, [kyberData, coingeckoData, isKyberDataNotValid, isReverse])
-
-  const error = (kyberError && coingeckoError) || (coingeckoData && chartData.length === 0)
-
-  const { data: liveKyberData } = useSWR(
-    !isKyberDataNotValid && kyberData && chainId
-      ? liveDataApi[chainId] + `?ids=${tokenAddresses[0]},${tokenAddresses[1]}`
-      : null,
-    fetchKyberDataSWR,
-    {
-      refreshInterval: 60000,
-      shouldRetryOnError: false,
-      revalidateOnFocus: false,
-      revalidateIfStale: false,
-    },
-  )
-  const { data: liveCoingeckoData } = useSWR(
-    isKyberDataNotValid && coingeckoData ? [tokenAddresses, chainId, 'live'] : null,
-    fetchCoingeckoDataSWR,
-    {
-      refreshInterval: 60000,
-      shouldRetryOnError: false,
-      revalidateOnFocus: false,
-      revalidateIfStale: false,
-    },
-  )
-  const latestData = useMemo(() => {
-    if (isKyberDataNotValid) {
-      if (liveCoingeckoData) {
-        const [data1, data2] = liveCoingeckoData
-        if (data1.prices?.length > 0 && data2.prices?.length > 0) {
-          const newValue = parseFloat(
-            (data1.prices[data1.prices.length - 1][1] / data2.prices[data2.prices.length - 1][1]).toPrecision(6),
-          )
-          return { time: new Date().getTime(), value: newValue }
-        }
-      }
-    } else {
-      if (liveKyberData) {
-        const value =
-          liveKyberData && tokenAddresses[0] && tokenAddresses[1]
-            ? liveKyberData[tokenAddresses[0]]?.price / liveKyberData[tokenAddresses[1]]?.price
-            : 0
-        if (value) return { time: new Date().getTime(), value: value }
-      }
-    }
-    return null
-  }, [liveKyberData, liveCoingeckoData, isKyberDataNotValid, tokenAddresses])
-  return {
-    data: useMemo(() => (latestData ? [...chartData, latestData] : chartData), [latestData, chartData]),
-    error: error,
-    loading: chartData.length === 0 && !error,
-  }
-}
+import { useMemo } from 'react'

```

Comment:
why the whole file is mark as change @XiaoYhun 😂

Reply:
maybe some endline issues 😢 but you can see changes are darker color